dist/components/SideBar/ChatHistoryItem.js

```diff
@@ -40,10 +39,11 @@ export const ChatHistoryItem = memo(({ conversation, onHistoryItemClick, deleteC
     const handleEdit = (e) => {
         e.stopPropagation();
         if (editingTitle) {
-            setEditInputValue(optimisticTitle);
             setEditingTitle(false);
         }
         else {
+            // Set the edit input value to the current optimistic title when entering edit mode
+            setEditInputValue(optimisticTitle);
```

Comment:
Before we were initializing the input's value as the title prop, but that starts a "New chat" and then gets updated to the summary. That doesn't update the default value for the input, so the stale value was shown.

We don't need to set the value for the input until we are about to enter edit mode when we have the latest value.